src/rcc.rs

```diff
@@ -294,6 +300,59 @@ pub trait Reset: RccBus {
     unsafe fn reset_unchecked();
 }
 
+/// Frequency on bus that peripheral is connected in
+pub trait BusClock {
+    /// Calculates frequency depending on `Clock` state
+    fn clock(clocks: &Clocks) -> Hertz;
+}
+
+impl<T> BusClock for T
+where
+    T: RccBus,
+    T::Bus: BusClock,
+{
+    fn clock(clocks: &Clocks) -> Hertz {
+        T::Bus::clock(clocks)
+    }
+}
+
+/// Frequency on bus that timer is connected in
+pub trait BusTimerClock {
+    /// Calculates base frequency of timer depending on `Clock` state
+    fn timer_clock(clocks: &Clocks) -> Hertz;
+}
+
+impl<T> BusTimerClock for T
+where
+    T: RccBus,
+    T::Bus: BusTimerClock,
+{
+    fn timer_clock(clocks: &Clocks) -> Hertz {
+        T::Bus::timer_clock(clocks)
+    }
+}
+
+impl BusTimerClock for APB1R1 {
+    fn timer_clock(clocks: &Clocks) -> Hertz {
+        let pclk_mul = if clocks.ppre1 > 1 { 2 } else { 1 };
```

Comment:
need check

Reply:
Is this resolved?

Reply:
No, I can't check it.
I've just copied this from F4. But L4 can have other multipliers. Or it can always be 1.
In F4 you can find it in `RCC_DCKCFGR/TIMPRE` register. I can't find something similar for L4.